### ISSUE TITLE
[Thread-1.3-Cert] Failure in a 1.3 cert issue TC 1.9 [Reachability - Single BR - IPv6 infrastructure with DHCPv6]

### ISSUE THREAD

**Issue by suveshpratapa** [openthread/ot-br-posix#1457]
Hi, we are failing TC 1.9 in Thread 1.3 cert, which tests bi-directional reachability between the Thread network and infrastructure devices, using an IPv6 infrastructure with DHCP (SLAAC not enabled).

Our test setup builds the 1.3 border router like so (from [here](https://github.com/openthread/ot-reference-release/blob/main/script/otbr-setup.bash)):
```
'INFRA_IF_NAME=eth0'
'RELEASE=1'
'REFERENCE_DEVICE=1'
'BACKBONE_ROUTER=1'
'BORDER_ROUTING=1'
'NETWORK_MANAGER=0'
'NAT64=1'
'DNS64=1'
'DHCPV6_PD=0'
'WEB_GUI=0'
'REST_API=0'
"OTBR_OPTIONS=\"-DOT_THREAD_VERSION=1.3 -DOT_FULL_LOGS=ON -DOT_BORDER_ROUTING=ON -DOTBR_DUA_ROUTING=ON -DOT_DUA=ON -DOT_MLR=ON -DOTBR_DNSSD_DISCOVERY_PROXY=ON -DOTBR_SRP_ADVERTISING_PROXY=ON -DOT_SRP_CLIENT=ON -DOT_TREL=ON -DOT_RCP_RESTORATION_MAX_COUNT=100 -DOTBR_PACKAGE_VERSION=${OT_BR_POSIX_COMMIT_HASH} -DOT_PACKAGE_VERSION=${OPENTHREAD_COMMIT_HASH}\""
```

We seem to be failing all steps in this test, starting from where it expects our OTBR to act as a DHCP client to use a delegated prefix.  

This test apparently passes for a Nordic platform, but fails for the EFR platform, so we don't understand why an OTBR/POSIX feature such as this is affected.

I attached the test case description and the logs/report from both the failing and passing cases. Can I get some understanding of the expected behavior here, and if something is wrong with our border router config?

[TC_1_9.pdf](https://github.com/openthread/ot-br-posix/files/9103520/TC_1_9.pdf)
[Test_1_9_Failure_EFR.zip](https://github.com/openthread/ot-br-posix/files/9103524/Test_1_9_Failure_EFR.zip)
[Test_1_9_Failure_NCS.zip](https://github.com/openthread/ot-br-posix/files/9103591/Test_1_9_Failure_NCS.zip)

**Comment by librasungirl** [comment]
The route cause seems that test harness explicitly configure mesh local prefix as `fd00:db8::/64`, which conflicts with the on-link ULA prefix that derives from extpanid (`000db80000000000`) since https://github.com/openthread/openthread/pull/7695 , thus causing external route fails to be published into netdata, and then no RA PIO accordingly.

```
fe80::ba27:ebff:fe11:e0f4 Jul 11 17:19:45 rpi-112 otbr-agent[804]: 00:06:16.865 [W] BorderRouter--: Failed to publish external route fd00:db8:0:0::/64: InvalidArgs
```

Nordic seems built before https://github.com/openthread/openthread/pull/7695 and its ULA prefix is randomly generated as below
```
fe80::e65f:1ff:fe9e:e2b9 May  7 16:54:04 raspberrypi otbr-agent[9141]: 00:00:54.539 [I] BorderRouter--: Adding external route fd2e:fe65:7b28:0::/64
fe80::e65f:1ff:fe9e:e2b9 May  7 16:54:04 raspberrypi otbr-agent[9141]: 00:00:54.539 [I] BorderRouter--: Start advertising new on-link prefix fd2e:fe65:7b28:0::/64 on interface 2
fe80::e65f:1ff:fe9e:e2b9 May  7 16:54:04 raspberrypi otbr-agent[9141]: 00:00:54.540 [I] BorderRouter--: Send on-link prefix fd2e:fe65:7b28:0::/64 in PIO (preferred lifetime = 1800 seconds, valid lifetime = 1800 seconds)
fe80::e65f:1ff:fe9e:e2b9 May  7 16:54:04 raspberrypi otbr-agent[9141]: 00:00:54.540 [I] BorderRouter--: Send OMR prefix fdac:5d6:6891:1::/64 in RIO (valid lifetime = 1800 seconds)
```
Update Test Harness to configure a different default Mesh Local prefix or a different default extpanid will help, I think.

**Comment by suveshpratapa** [comment]
Also @librasungirl it appears this change was reverted? https://github.com/openthread/openthread/pull/7709

**Comment by jwhui** [comment]
> @librasungirl Can you provide guidance as to what the mesh-local prefix should be given the ext. pan id `000db80000000000`
> 
> Also, @jwhui do you know which part of the Thread spec mandates this?

See Thread 1.3.0-RC3 Section 9.6.2.

**Comment by librasungirl** [comment]
It looks like the pcap missed the BR DUT's thread traffic when boots (e.g. parent request, no response, then data response indicating it becomes leader etc), and the first mle message from BR DUT is the MLE advertisement with `data version 204 in leader data tlv`, same as what in Child ID Response message, which has the expecting prefix/route/services etc.

For thread network data check, may either 1) start the sniffer earlier or start BR DUT later to ensure BR's attachment process is captured, or 2) check more flexibly in either Data Response or Child ID Response.

For RA PIO/RIO check, if it assumes closely after MLE Data Response, may try option 1); otherwise may extend the packet range for RA check, or may explicitly initiate an RS to trigger the RA response to ensure there must be the RA in the expecting packet range.

**Comment by EskoDijk** [comment]
On the RA check, current specs (RFC 4861) suggest that the RA with the new prefix information should be sent within MAX_INITIAL_RTR_ADVERT_INTERVAL = 16 seconds. So the harness could also wait for this time (at most) to see the expected change in an RA message.   (Note this isn't specified as a MUST, so if a DUT fails to do this we may need to bring this back to TC discussion if we need to specify that. But I expect DUTs will do this.)

**Comment by librasungirl** [comment]
agree. but in current test the sniffer seems not starting along with BR DUT's boot, there might be chance that the first 16s period of announcing this new prefix is missed. overall I think the option 1 will make things easier.

---
another concern on nordic reference is that it seems still thread version 3 in the parent response, should be updated to be 4 as a thread 1.3 reference.

**Comment by suveshpratapa** [comment]
> another concern on nordic reference is that it seems still thread version 3 in the parent response, should be updated to be 4 as a thread 1.3 reference.

This should be fixed after this PR is merged: https://github.com/openthread/ot-reference-release/pull/34
cc @jwhui @edmont

**Comment by edmont** [comment]
> This should be fixed after this PR is merged: [openthread/ot-reference-release#34](https://github.com/openthread/ot-reference-release/pull/34)

I'm afraid this won't be enough for NCS, but I'm preparing the needed changes.

**Comment by jwhui** [comment]
Closing stale issue.
